Wallet patch (#390)
Agoric/documentation: main/wallet-api.md

  exposed via the iframe/WebSocket bridge that a Dapp UI can use to access the
  wallet.
  
## WalletUser API commands

### `getBridge()`
- Returns: `{Promise<WalletBridge>}`

Returns the wallet bridge that bypasses Dapp-authorization. This should 
only be used within the REPL or deployment scripts that want to use the
WalletBridge API without the effort of calling `getScopedBridge`.
**tyg todo: Why is this an effort? How severe is it that we're recommedning 
they not do it?**

### `getScopedBridge(suggestedDappPetname, dappOrigin)`
- `suggestedDappPetname` `{Petname}`
- `dappOrigin` `{String}`
- Returns: `{Promise<WalletBridge>}`

Returns a wallet bridge corresponding to an origin that must be approved in the wallet UI. 
This is available for completeness to provide the underlying API that's available over the
standard wallet-bridge.html.

### `addPayment(payment)`
- `id` `{ERef<Payment>}`
- Returns: `void`

Adds a payment to the Wallet for deposit to the user-specified purse,
either via an autodeposit or manually approved.

### `getDepositFacet(brandBoardId)`
- `brandBoardId` `{String}`
- Returns: `{Promise<string>}`

> Returns the board ID for the deposit facet of a purse in the user's Wallet that accepts payments
> of the brand specified by the `brandBoardId` parameter. **tyg todo: What if
> the Wallet has more than one purse for the specified by board ID brand? Or is
> the parameter really the id for a purse that takes that brand?**

Reviewer: ```suggestion
Returns the board ID for the deposit facet of the user's Wallet that accepts payments
of the brand specified by the `brandBoardId` parameter.
```

### `getIssuers()`
- Returns: `{Array<[Petname, Issuer]>}`

Returns an array of all the Issuers and their petnames associated with this Wallet.

### `getIssuer(petname)`
- `petname` `{Petname}`
- Returns: `{Issuer}`

Returns the issuer with the specified petname associated with this Wallet.

### `getPurses()`
- Returns: `{Array<[Petname, Purse]>}`

Returns all the purses associated with this wallet.

### `getPurse(pursePetname)`
- `pursePetName`  `{String}`
- Returns `{Purse}`
- Errors: Throws an error if there is no purse with the given petname.

Returns the `purse` object with the given petname
    
## WalletBridge API commands    
    
These methods can be used by an untrusted Dapp without breaching the wallet's 
integrity.  They are also exposed via the iframe/WebSocket bridge that a 
Dapp UI can use to access a Wallet.
 
### `addOffer(offer)`
- `offer` `{OfferState}`
- Returns: `{Promise<string>}`

> Adds an offer to the Wallet, returning the offer's unique Board ID. 

Reviewer: ```suggestion
Adds an offer to the Wallet, returning the offer's unique private ID in the Wallet.  This ID is not stored in the Board.
```

### `addOfferInvitation(offer, invitation)`
- `offer` `{OfferState}`
- `invitation` `{ERef<Payment>}`
- Returns: `{Promise<string>}`

> Add the specified invitation to the specified offer, returning the offer's unique Board ID. 

Reviewer: ```suggestion
Add the specified invitation to the specified offer, returning the offer's private ID in the Wallet.  This ID is not stored in the Board.
```

![Make Purse](./assets/MakePurseWalletUI.png)

The Issuer creates a new empty Purse, that holds its Brand of assets, in the Wallet, giving it the Petname
you specify. Remember there can be more than one Purse in a Wallet that holds assets of a specific Brand.

If you click the **Import** button at the bottom of the Issuers list, this popup appears:

![Import Issuer](./assets/ImportIssuerWalletUI.png)

You specify a Petname and the Board ID (obtained from a trusted source) of an Issuer, and it's imported
into the Wallet and can be used to create new empty Purses to store assets of its associated Brand. 

### Contacts

![Contacts](./assets/ContactsWalletUI)

The Contacts component shows all entities known to the Wallet, including the Wallet itself as "Self". An
expanded entry shows the contact's Board ID. If you click on the **Import** button, this popup appears:

![Import Contact](./assets/ImportContactWalletUI)

You specify a Petname and the Board ID (obtained from a trusted source) of a Contact, and it's imported
into the Wallet. 

### Offers

**tyg todo: Not sure what this looks like as default Faucet Wallet doesn't list any.
Need to rework things **

### Incoming Payments

> **tyg todo: Not sure what this looks like as default Faucet Wallet doesn't list any**

Author: Whether I have auto-deposit enabled or not, payments still seem to zip through and deposit themselves, blinking up something under incoming payments that disappears before I can screen capture.

### `getIssuers()`
- Returns: `{Array<[Petname, Issuer]>}`

Returns an array of all the Issuers and their petnames associated with this Wallet.

### `getIssuer(petname)`
- `petname` `{Petname}`
- Returns: `{Issuer}`

Returns the issuer with the specified petname associated with this Wallet.

### `getPurses()`
- Returns: `{Array<[Petname, Purse]>}`

Returns all the purses associated with this wallet.

### `getPurse(pursePetname)`
- `pursePetName`  `{String}`
- Returns `{Purse}`
- Errors: Throws an error if there is no purse with the given petname.

Returns the `purse` object with the given petname
    
## WalletBridge API commands    
    
These methods can be used by an untrusted Dapp without breaching the wallet's 
integrity.  They are also exposed via the iframe/WebSocket bridge that a 
Dapp UI can use to access a Wallet.
 
### `addOffer(offer)`
- `offer` `{OfferState}`
- Returns: `{Promise<string>}`

Adds an offer to the Wallet, returning **tyg todo: Not sure what the returned string is for**

### `addOfferInvitation(offer, invitation)`
- `offer` `{OfferState}`
- `invitation` `{ERef<Payment>}`
- Returns: `{Promise<string>}`

> Add the specified invitation to the specified offer. **tyg todo: Not sure what the returned string is for**

Reviewer: ```suggestion
Add the specified invitation to the specified offer, returning the offer's unique Id.
```

### `getDepositFacetId(brandBoardId`
- `brandBoardId` `{string}`
- Returns: `{Promise<string>}`

Returns the Board ID to use to receive payments of the specified by its Board ID brand.

### `getPursesNotifier()`
- Returns: `{Promise<Notifier<Array<PursesFullState>>>}`

Returns a notifier that follows changes the purses in the Wallet.

### `getOffersNotifier()`
- Returns: `{Promise<Notifier<array<OfferState>>>}`

Returns a notifier that follows changes to the offers received by the Wallet.

### `suggestIssuer(petname, issuerBoardId)`
- `petname` `{Petname}`
- `issuerBoardId` `{string}`
- Returns: `void`

Introduce an ERTP issuer with a suggested petname to the Wallet.

### `suggestInstallation(petname, installationBoardID)`
- `petname` `{Petname}`
- `installationBoardId` `{string}`
- Returns: `void`

Introduce a Zoe contract installation with a suggested petname to the Wallet.

### `suggestInstance(petname, instanceBoardId)
- `petname` `{Petname}`
- `instanceBoardId` `{string}`
- Returns: `void`

Introduce a Zoe contract instance with a suggested petname to the Wallet.
 

> **tyg todo: old API stuff starts here. Should any of it be retained?**

Reviewer: No, none should be retained.

Please move the rest to a separate, unlinked document in case we want to revive any of it when we make the APIs public again.
